src/main/scala/firrtl/PrimOps.scala

```diff
@@ -159,42 +172,48 @@ object PrimOps extends LazyLogging {
         case (_: UIntType, _: UIntType) => Utils.BoolType
         case (_: SIntType, _: SIntType) => Utils.BoolType
         case (_: FixedType, _: FixedType) => Utils.BoolType
+        case (_: IntervalType, _: IntervalType) => Utils.BoolType
```

Comment:
Can these be ors? This is getting out of hand
```
case (_: UIntType, _: UIntType) | (_: SIntType, _: SIntType) | ... => Utils.BoolType
case _ => UnknownType
```

Reply:
I'm not sure that's actually better? It can be ors, but then that line will be too long and you'd have to wrap it anyways